### PR TITLE
Rename Filesystem to FileSystem

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okio/src/appleMain/kotlin/okio/posixVariant.kt [hunk 1]
```diff
@@ -25,8 +25,8 @@ import platform.posix.S_IFREG
 import platform.posix.errno
 import platform.posix.stat
 
-@ExperimentalFilesystem
-internal actual fun PosixSystemFilesystem.variantMetadataOrNull(path: Path): FileMetadata? {
+@ExperimentalFileSystem
+internal actual fun PosixSystemFileSystem.variantMetadataOrNull(path: Path): FileMetadata? {
```
Comment (Egorand) [inline okio/src/appleMain/kotlin/okio/posixVariant.kt:29]: nit: any reason this isn't just `PosixFileSystem`?

File: okio/src/jsMain/kotlin/okio/-Platform.kt [hunk 2]
```diff
@@ -17,11 +17,11 @@ package okio
 
 import okio.Path.Companion.toPath
 
-@ExperimentalFilesystem
-internal actual val PLATFORM_FILESYSTEM: Filesystem
-  get() = NodeJsSystemFilesystem
+@ExperimentalFileSystem
+internal actual val PLATFORM_FILESYSTEM: FileSystem
```
Comment (Egorand) [inline okio/src/jsMain/kotlin/okio/-Platform.kt:21]: `PLATFORM_FILE_SYSTEM`?

File: okio/src/jsMain/kotlin/okio/NodeJsSystemFileSystem.kt [hunk 3]
```diff
@@ -18,15 +18,15 @@ package okio
...
  *
  * This class needs to make calls to some fs APIs that have multiple competing overloads. To
  * unambiguously select an overload this passes `undefined` as the target type to some functions.
  *
  * [node_fs]: https://nodejs.org/dist/latest-v14.x/docs/api/fs.html
  */
-@ExperimentalFilesystem
-internal object NodeJsSystemFilesystem : Filesystem() {
+@ExperimentalFileSystem
+internal object NodeJsSystemFileSystem : FileSystem() {
```
Comment (Egorand) [inline okio/src/jsMain/kotlin/okio/NodeJsSystemFileSystem.kt:29]: nit: same re `NodeJsFileSystem`

**Review by Egorand — APPROVED** [review]

File: okio/src/appleMain/kotlin/okio/posixVariant.kt (hunk 1 above)
Comment (swankjesse) [reply okio/src/appleMain/kotlin/okio/posixVariant.kt:29]: Lemme make that change!

File: okio/src/jsMain/kotlin/okio/-Platform.kt (hunk 2 above)
Comment (swankjesse) [reply okio/src/jsMain/kotlin/okio/-Platform.kt]: Fixed!

File: okio/src/jsMain/kotlin/okio/NodeJsSystemFileSystem.kt (hunk 3 above)
Comment (swankjesse) [reply okio/src/jsMain/kotlin/okio/NodeJsSystemFileSystem.kt]: Fixed.

**Pull request merged**